sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -537,6 +539,19 @@ case class ApplyColumnarRulesAndInsertTransitions(
           false
       }
       plan.withNewChildren(plan.children.map(insertTransitions(_, outputsColumnar)))
+    } else {
```

Comment:
The if-else clause

```scala
if (plan.isInstanceOf[ColumnarToRowTransition]) {
  ...
} else {
  ...
}
```

is inverted here to ensure symmetry with the code in `ensureOutputsColumnar`.